packages/babel-traverse/src/path/family.ts

```diff
@@ -193,11 +193,16 @@ function getStatementListCompletion(
     }
   } else if (paths.length) {
     // When we are in a context where `break` must not exist, we can skip linear
-    // search on statement lists and assume that the last statement determines
-    // the completion
-    completions = completions.concat(
-      _getCompletionRecords(paths[paths.length - 1], context),
-    );
+    // search on statement lists and assume that the last
+    // non-variable-declaration statement determines the completion.
+    for (let i = paths.length - 1; i >= 0; i--) {
+      if (!paths[i].isVariableDeclaration()) {
```

Comment:
I think we should apply `_getCompletionRecords` and check if the completion is `VariableDeclaration`, e.g. The completion record of this snippet

```js
1 + 2;
{
  var a = 1;
}
```
is `1 + 2`.

Reply:
@JLHwung Sure, done, and added a test. I have to admit that I’m a bit unsure about the code here – is it valid to assume that if the list of completion records returned by `_getCompletionRecords` contains a `VariableDeclaration`, then that automatically implies that the list has length 1? It seems like something that should be true to me, and I’m not sure how to handle this if it isn’t.

Reply:
> is it valid to assume that if the list of completion records returned by _getCompletionRecords contains a VariableDeclaration, then that automatically implies that the list has length 1

That's a very good question! AFAIK only IfStatement, SwitchStatement and TryStatement may produce multiple completion records, and if a VariableDeclaration is directly within these statements, it should be recursively handled here and as long as we don't return VariableDeclaration as completion records in `getStatementListCompletion`, then the `VariableDeclaration` must _not_ come from these statements and thus the list here should contain only one completion record.

Reply:
Right – that’s why I’d be thinking. :) Should I add an assertion of some kind?

Reply:
> an assertion of some kind

I think it's fine. The `_getCompletionRecords` is mostly used in `proposal-do-expression`, and the proposal now [bans](https://tc39.es/proposal-do-expressions/#sec-doexpression-static-semantics-early-errors) variable declaration ending in do block. So I guess few people will use it in that way.